app/src/main/java/com/example/pix/chat/fragments/MusicRoomFragment.java

```diff
@@ -127,77 +198,50 @@ private void seekSong() {
                         Toast.makeText(getContext(), "Error updating MusicRoom!", Toast.LENGTH_SHORT).show();
                         return;
                     }
-                    setup();
+                    startSync();
                 });
             });
             // If the user is a listener and there is no Song, they must wait until the MusicRoom has a Song
-        } else
-            new Timer().schedule(new TimerTask() {
-                @Override
-                public void run() {
-                    try {
-                        musicRoom = q.getFirst();
-                        nowPlayingInParse = musicRoom.getCurrentSong();
-                        if (nowPlayingInParse != null) {
-                            setup();
-                            this.cancel();
-                        }
-                    } catch (ParseException e) {
-                        e.printStackTrace();
-                        Toast.makeText(getContext(), "Error looking for new Song!", Toast.LENGTH_SHORT).show();
-                        this.cancel();
-                    }
+        } else {
+            try {
+                musicRoom = q.getFirst();
```

Comment:
Why do you need to do this now? Can't you just set the `musicRoom` when the query was first executed?

Reply:
I need to keep running the query to "refresh" the musicroom because at this point it is null, meaning our original query returned no objects because this friend we are attempting to listen to is not streaming. I made it so that you can keep pressing the play button to refresh and try again.

Reply:
I see.